src/backend/distributed/executor/multi_utility.c

```diff
@@ -613,11 +613,48 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 					}
 
 					constraint = (Constraint *) command->def;
-					if (ConstraintIsAForeignKey(constraint->conname, relationId))
+					if (constraint->contype == CONSTR_FOREIGN)
 					{
 						InvalidateForeignKeyGraph();
 					}
 				}
+				else if (alterTableType == AT_AddColumn)
+				{
+					List *columnConstraints = NIL;
+					ListCell *columnConstraint = NULL;
+					Oid relationId = InvalidOid;
+					LOCKMODE lockmode = NoLock;
+
+					ColumnDef *columnDefinition = (ColumnDef *) command->def;
+					columnConstraints = columnDefinition->constraints;
+					if (columnConstraints)
+					{
+						ErrorIfUnsupportedAlterAddConstraintStmt(alterTableStatement);
+					}
+
+					lockmode = AlterTableGetLockLevel(alterTableStatement->cmds);
+					relationId = AlterTableLookupRelation(alterTableStatement, lockmode);
+					if (!OidIsValid(relationId))
+					{
+						continue;
+					}
+
+					foreach(columnConstraint, columnConstraints)
+					{
+						Constraint *constraint = (Constraint *) lfirst(columnConstraint);
+
+						if (constraint->conname == NULL &&
+								(constraint->contype == CONSTR_PRIMARY ||
```

Comment:
is this better than allowing a set of commands that are known to work ?

Reply:
Which commands that we should allow ? The commands that we avoid here are the ones that we can create but cannot drop. I think not allowing such commands is better than allowing the creation of nondestroyable constraints in a straightforward manner

Reply:
tt2 is a reference table, it has column b of bigserial type

following is a valid for a regular table, but fails with a worker error. 

```sql
 alter table tt2 add column d  bigint default nextval('tt2_b_seq'::regclass);
ERROR:  relation "tt2_b_seq" does not exist
CONTEXT:  while executing command on localhost:9703
```
It is not a big issue and nowhere real being showstopper. just a nuisance;